calico_containers/integrations/kubernetes/tests/kube_plugin_test.py

```diff
@@ -0,0 +1,378 @@
+            m_interface.assert_called_once_with()
+            m_prof.assert_called_once_with('endpt_id')
+
+    def test_create_fail(self):
+        with patch.object(self.plugin, '_configure_interface') as m_interface, \
+                patch.object(self.plugin, '_configure_profile') as m_prof, \
+                patch('calico_containers.integrations.'
+                      'kubernetes.calico_kubernetes.sys.exit') as m_sys:
+            # Set up mock objects
+            m_interface.side_effect = CalledProcessError(1,'','')
+
+            # Set up args
+            args = ['script name','setup','','podname','dockerId']
+
+            # Call method under test
+            self.plugin.create(args)
+
+            # Assert
+            m_sys.assert_called_once_with(1)
+
+    def test_delete(self):
+        with patch.object(self.plugin, 'calicoctl') as m_calicoctl:
+            # Set up args
+            args = ['script name','teardown', '','pod-name','dockerId']
+
+            # Call method under test
+            self.plugin.delete(args)
+
+            # Assert
+            self.assertEqual(self.plugin.pod_name, 'pod_name')
+            self.assertEqual(self.plugin.docker_id, 'dockerId')
+            m_calicoctl.assert_any_call('container', 'remove', 'dockerId')
+            m_calicoctl.assert_any_call('profile', 'remove', 'pod_name')
+
+    def test_configure_interface(self):
+        with patch.object(self.plugin, '_read_docker_ip') as m_read_docker, \
+                patch.object(self.plugin, '_delete_docker_interface') as m_delete_docker, \
+                patch.object(calico_kubernetes, 'container_add') as m_add_container, \
+                patch.object(calico_kubernetes, 'generate_cali_interface_name') as m_cali, \
+                patch.object(self.plugin, '_get_node_ip') as m_node_ip, \
+                patch.object(calico_kubernetes, 'check_call') as m_check_call:
+            # Set up mock objects
+            m_read_docker.return_value = 'docker_ip'
+            class ep:
+                endpoint_id = 'ep_id'
+            m_add_container.return_value = ep
+            m_cali.return_value = 'interface_name'
+            m_node_ip.return_value = 'node_ip'
+
+            # Call method under test
+            return_val = self.plugin._configure_interface()
+
+            # Assert
+            m_read_docker.assert_called_once()
+            m_delete_docker.assert_called_once()
+            m_add_container.assert_called_once_with(
+                self.plugin.docker_id, 'docker_ip', 'eth0')
+            m_cali.assert_called_once_with(IF_PREFIX,'ep_id')
+            m_node_ip.assert_called_once()
+            m_check_call.assert_called_once_with(
+                ['ip', 'addr', 'add', 'node_ip' + '/32',
+                'dev', 'interface_name'])
+            self.assertEqual(return_val.endpoint_id, 'ep_id')
+
+    def test_get_node_ip(self):
+        with patch('calico_containers.integrations.kubernetes.'
```

Comment:
Make sure all of your patch calls use the "autospec=true" argument.

This is useful because it make sure the returned Mock object uses the same spec as the object being mocked.